src/js/common/schemaform/widgets/ArrayCountWidget.jsx

```diff
@@ -1,45 +1,45 @@
 import React from 'react';
 
-// We're expanding or contracting the array based on the count
-// and returning undefined if the array should be empty
-function getValue(count, value = []) {
-  if (!count || parseInt(count, 10) < 1) {
-    return undefined;
+export default class SpouseMarriageWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: !!props.value
+      ? props.value.length - (this.props.options.countOffset || 0)
+      : undefined };
   }
 
-  const intCount = parseInt(count, 10);
-  if (intCount < value.length) {
-    return value.slice(0, intCount);
-  }
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.value !== this.state.value) {
+      let count = parseInt(this.state.value, 10);
+      if (isNaN(count)) {
+        count = 0;
+      }
 
-  return value.concat(Array(intCount - value.length).fill({}));
-}
+      if (count > 29) {
+        count = 29;
+      }
 
-export default class ArrayCountWidget extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { value: !!props.value ? props.value.length : undefined };
+      this.props.onChange(this.getValue(count, this.props.value));
+    }
   }
+  // We're expanding or contracting the array based on the count
+  // and returning undefined if the array should be empty
+  getValue = (count, value = []) => {
+    if (count === 0) {
+      return undefined;
+    }
+
+    const intCount = count + (this.props.options.countOffset || 0);
 
-  // Need to keep the count state in sync, but don't want to overwrite if the
-  // user has chosen '' or 0
-  componentWillReceiveProps(props) {
-    const arrayData = props.value || [];
-    if (!!arrayData.length && (!this.state.value || this.state.value !== arrayData.length)) {
-      this.setState({ value: arrayData.length });
+    if (intCount < value.length) {
+      return value.slice(0, intCount);
     }
+
+    return Array(intCount - value.length).fill({}).concat(value);
```

Comment:
Yeah, I have to guess here, and I picked this way because the last one is titled "current marriage" and seems like that's the one they've mostly likely entered correctly.